### PR TITLE
Correct behaviour to the Leave and Unlock button

### PR DESCRIPTION
**Pull request by richrace**

### CODE REVIEW

File: src/features/CourtCaseList/CourtCaseList.tsx [hunk 1]
```diff
@@ -20,6 +20,18 @@ const CourtCaseList: React.FC<Props> = ({ csrfToken, courtCases, order = "asc",
   const recentlyUnlockedExceptionId = query.unlockException
   const recentlyUnlockedTriggerId = query.unlockTrigger
 
+  const queryString = Object.entries(query)
+    .reduce((acc, [key, value]) => {
+      if (key === "unlockException" || key === "unlockTrigger") {
```
Comment (C-gyorfi) [inline src/features/CourtCaseList/CourtCaseList.tsx:25]:
```
      if (key !== "unlockException" && key !== "unlockTrigger") {
            acc.push(`${key}=${value}`)
      }
      return acc
```

What do you think?

**Review by C-gyorfi — APPROVED** [review]

File: src/features/CourtCaseList/CourtCaseList.tsx (hunk 1 above)
Comment (richrace) [reply src/features/CourtCaseList/CourtCaseList.tsx:25]: I debated that myself but preferred this. In my opinion, this is more readable. I find it tricky negating values (more cognitive load).

@richardcane have a preference?

Comment (C-gyorfi) [reply src/features/CourtCaseList/CourtCaseList.tsx:25]: @richrace I don't have a strong preference, happy for you to merge it!

File: src/pages/court-cases/[courtCaseId]/resolve.tsx [hunk 2]
```diff
@@ -77,7 +85,13 @@ export const getServerSideProps = withMultipleServerSideProps(
...
       )
 
-      return redirectTo(`/court-cases/${courtCase.errorId}`)
+      let redirectUrl = `/court-cases/${courtCase.errorId}`
+
+      if (previousPath) {
+        redirectUrl += `?previousPath=${encodeURIComponent(previousPath)}`
+      }
+
+      return redirectTo(redirectUrl)
```
Comment (richrace) [inline src/pages/court-cases/[courtCaseId]/resolve.tsx:94]: This might just be:

```suggestion
      return redirectTo(`/`)
```

**Pull request merged**